File: api/schemas/shared_structures/categorized_summary/summary_category.py
import graphene


class SummaryCategory(graphene.ObjectType):
    """
    This object contains the information for each type of summary that has been
    pre-computed
    """

    name = graphene.String(
        description="Category of computed summary which the other fields relate to."
    )
    count = graphene.Int(
        description="Total count of domains that fall into this category."
    )
    percentage = graphene.Float(description="Percentage compared to other categories.")

    def resolve_name(self, info, **kwargs):
        return self.name

    def resolve_count(self, info, **kwargs):
        return self.count

    def resolve_percentage(self, info, **kwargs):
        return self.percentage

File: api-js/src/types/categorized-summary/summary-category.js
const {
  GraphQLObjectType,
  GraphQLString,
  GraphQLInt,
  GraphQLFloat,
} = require('graphql')

const summaryCategoryType = new GraphQLObjectType({
  name: 'SummaryCategory',
  fields: () => ({
    name: {
      type: GraphQLString,
      description: `Category of computed summary which the other fields relate to.`,
      resolve: async () => {},
    },
    count: {
      type: GraphQLInt,
      description: `Total count of domains that fall into this category.`,
      resolve: async () => {},
    },
    percentage: {
      type: GraphQLFloat,
      description: `Percentage compared to other categories.`,
      resolve: async () => {},
    },
  }),
  description: `This object contains the information for each type of summary that has been pre-computed`,
})

module.exports = {
  summaryCategoryType,
}
